src/utils/manageLocalStorage.ts

```diff
@@ -1,22 +1,20 @@
 import { GroupInfo } from "./globalTypes";
 
-const getLocalGroups = (): GroupInfo[] | [] => {
+const getLocalGroups = (): GroupInfo[] => {
   // push local storage to the group state
-  const res: string =
-    localStorage.getItem(import.meta.env.VITE_LOCALLY_STORED_GROUPS) ||
-    "undefined";
+  const res = localStorage.getItem(import.meta.env.VITE_LOCALLY_STORED_GROUPS)
 
   // exit function if there was no local storage
-  if (res === "undefined") return [];
+  if (!res) return [];
```

Comment:
I wouldn't coalesce into a string that says `"undefined"` and then expeclity check for a string that says `"undefined"`

If it doesn't return a value, you can just check for that return value explicitly